hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutput.java

```diff
@@ -252,7 +256,7 @@ private synchronized void failed(Channel channel, Supplier<Throwable> errorSuppl
     // disable further write, and fail all pending ack.
     state = State.BROKEN;
     failWaitingAckQueue(channel, errorSupplier);
-    datanodeInfoMap.keySet().forEach(ChannelOutboundInvoker::close);
+    datanodeInfoMap.keySet().forEach(NettyFutureUtils::safeClose);
```

Comment:
I like the approach here where new `NettyFutureUtils` encapsulates the await/cleanup details.